godot-core/src/task/async_runtime.rs

```diff
@@ -87,25 +348,101 @@ use crate::private::handle_panic;
 /// });
 /// ```
 #[doc(alias = "async")]
-pub fn spawn(future: impl Future<Output = ()> + 'static) -> TaskHandle {
-    // Spawning new tasks is only allowed on the main thread for now.
+pub fn spawn(future: impl Future<Output = ()> + Send + 'static) -> TaskHandle {
+    // Check if runtime is registered
+    if !is_runtime_registered() {
+        panic!("No async runtime has been registered. Call gdext::task::register_runtime() before using async functions.");
+    }
```

Comment:
This is a breaking change.

So far, could use the "Godot runtime" as a default. That one supports some features (signal awaiting) but not all (`async fn`).

Do you think it's possible to still provide that as a default, and later let the user explicitly select which one to use?